### PR TITLE
Add switch to disable transaction resubmission

### PR DESCRIPTION
**Pull request by wltsmrz**
* Add Transaction.setResumittable(Boolean)
* Add Transaction.isResumittable() -> Boolean
* Add automatic_resubmission (Boolean) config option to Remote

### CODE REVIEW

File: src/core/transactionmanager.js [hunk 1]
```diff
@@ -476,6 +476,11 @@ TransactionManager.prototype._resubmit = function(ledgers_, pending_) {
       }
     }
 
+    if (!transaction.isResubmittable()) {
+      // Allow the transaction to fail via ledger close
```
Comment (clark800) [inline src/core/transactionmanager.js]: I'm having trouble understanding this comment, could you add more explanation to it?

Comment (wltsmrz) [inline src/core/transactionmanager.js]: Ye

File: test/transaction-test.js [hunk 2]
```diff
@@ -1008,6 +1008,22 @@ describe('Transaction', function() {
...
+    assert.strictEqual(tx.isResubmittable(), true);
+
+    tx.setResubmittable(false);
+    assert.strictEqual(tx.isResubmittable(), false);
+
+    tx.setResubmittable(true);
+    assert.strictEqual(tx.isResubmittable(), true);
+
+    tx.setResubmittable(false);
+    tx.setResubmittable();
```
Comment (clark800) [inline test/transaction-test.js]: Can we make it require a boolean to simplify the type annotation that we'll add eventually?

Comment (wltsmrz) [inline test/transaction-test.js]: Sure

**Comment by clark800** [comment]
LGTM

**Pull request merged**